### ISSUE TITLE
@Field decorator with type instead of class

### ISSUE THREAD

**Issue by lj-vpd** [MichalLytek/type-graphql#1349]
**Describe the issue**
Is it possible, somehow, to use the `@Field` decorator with a type instead of a class?

We're trying to reduce circular dependency issues by changing our TypeORM decorators [like so](https://github.com/typeorm/typeorm/issues/4526#issuecomment-1076462998), but this work won't have an effect if we can't also take the same approach with TypeGraphQL, given our entity classes have decorators for TypeORM and TypeGraphQL.

**Are you able to make a PR that fix this?**
Possibly, with some more context.

**Additional context**
N/A

**Comment by lj-vpd** [comment]
Thanks for the quick response

And there's no other way this could work without passing a class?

**Comment by MichalLytek** [comment]
No, decorators = classes. Doesn't the `() => Foo` delayed syntax fix circular dependency issues?

**Comment by lj-vpd** [comment]
> No, decorators = classes

Sorry, my question is whether the decorator can _accept_ a string instead of a class (not whether or not the decorator itself is a class)

> Doesn't the () => Foo delayed syntax fix circular dependency issues?

In our case, apparently not

The `@ManyToOne` (TypeORM) decorator appears to accept strings, so I was just wondering if the same might apply here
